src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -242,16 +244,77 @@ export class DefaultTelemetryService implements TelemetryService {
         return event
     }
 
-    private static readEventsFromCache(cachePath: string): TelemetryEvent[] {
+    public static readEventsFromCache(cachePath: string): TelemetryEvent[] {
```

Comment:
Instead of changing the scope of readEventsFromCache, can filterTelemetryCacheEvents be made into a global standalone method that is tested in isolation?

Reply:
Sure we can do it that way, changed